keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -172,39 +177,88 @@ Kubelet components.
+
+#### Policies
+
+**none (default)**: Kubelet does not consult Topology Manager for placement decisions. 
+
+**best-effort**: Topology Manager will provide the preferred allocation for the container based
+on the hints provided by the Hint Providers. If an undesirable allocation is determined, the pod will be admitted with this undesirable allocation.
+
+**restricted**: Topology Manager will provide the preferred allocation for the container based
+on the hints provided by the Hint Providers. If an undesirable allocation is determined, the pod will be rejected. 
+This will result in the pod being in a `Terminated` state, with a pod admission failure.
+
+**single-numa-node**: Topology mananager will enforce an allocation of all resources on a single NUMA Node. If such
+an allocation is not possible, the pod will be rejected. This will result in the pod being in a `Terminated` state, with a pod admission failure.
 
 The Topology Manager component will be disabled behind a feature gate until
 graduation from alpha to beta.
 
 #### Computing Preferred Affinity
 
-A topology hint indicates a preference for some well-known local resources.
-Initially, the only supported reference resource is a mask of CPU socket IDs.
-After collecting hints from all providers, the Topology Manager chooses some
-mask that is present in all lists. Here is a sketch:
+After collecting hints from all providers, the Topology Manager preforms the
+affinity calcuation to determine the best fit Topology Hint.
+
+The chosen Topology Manager policy then decicds to admit or reject the pod based on this hint.
 
-1. Apply a partial order on each list: number of bits set in the
-   mask, ascending. This biases the result to be more precise if
-   possible.
-1. Iterate over the permutations of preference lists and compute
-   bitwise-and over the masks in each permutation.
-1. Store the first non-empty result and break out early.
-1. If no non-empty result exists, return an error.
+**Affinity Calcuation:**
+1. Loops through the list of hint providers and saves an accumulated list of 
+   the hints returned by each hint provider.
+2. Iterates through all permutations of hints accumulated in Step 1. The hint affinites are merged to a single hint
+   by performing a bitwise AND. The preferred field on the merged hint is set to false if any of the hints in the 
+   permutation returned a false preferred.
+3. The hint with the narrowest preferred affinity is returned.
+   * Narrowest in this case means the least number of NUMA nodes required to satisfy the resource request.      
+4. If no hint with at least one NUMA Node set is found, return a default hint which is a hint
+   with all NUMA Nodes set and preferred set to false.
 
-The behavior when a match does not exist is configurable, as described
-above.
+**Policy Decisions:**
```

Comment:
I know this is a slight repeat of the above information on the policies but I think it gives a good indication of what exactly is happening during the execution and the above is more of a broad explanation